src/app/qgsattributetabledialog.cpp

```diff
@@ -762,7 +765,46 @@ void QgsAttributeTableDialog::mActionCutSelectedRows_triggered()
 
 void QgsAttributeTableDialog::mActionCopySelectedRows_triggered()
 {
-  QgisApp::instance()->copySelectionToClipboard( mLayer );
+  // TODO: Loop selected rows
+  if ( mMainView->view() == QgsDualView::AttributeTable )
+  {
+    QList<QgsFeatureId> featureIds = mMainView->tableView()->selectedFeaturesIds();
+    QgsFeatureStore featureStore;
+    QgsFields fields = QgsFields( mLayer->fields() );
+    QStringList fieldNames;
+
+    Q_FOREACH ( const QgsAttributeTableConfig::ColumnConfig &columnConfig, mMainView->attributeTableConfig().columns() )
```

Comment:
Someone decided Q_FOREACH is no longer cool, so now the cool code is more lines:

```
const auto configs = mMainView->attributeTableConfig().columns();
for ( const QgsAttributeTableConfig::ColumnConfig &columnConfig : configs )
```